### PR TITLE
Local-first storage: repo-owned logs, derived global index (#48)

### LINKED ISSUE #2: creates .timescope folder even if not used
When I open a folder a .timescope folder is always created. Even if I am not running TimeScope within the folder.
The .timescope folder should not be created until someone chooses to start TimeScope within the folder.
Additionally, if there is no .timescope folder local then a prompt should come up "Log in this local folder too?" the user has to confirm this in order for the .timescope folder to be created.

Edge case
If a .timescope folder already exists then assume you should log locally and globally.

### LINKED ISSUE #48: Local-first storage architecture: repo-owned logs, derived global index
## Problem

The current storage model is **two coequal writable stores** (global `logs.jsonl` + workspace `.timescope/logs.jsonl`) with dual writes and merge/dedup. No event has a single owner. This is the root cause behind most of the focus milestone: #47 (multi-instance corruption), #2 (speculative folder creation), #3 (source confusion), #6 (local jobs question).

## Decision (made with user, 2026-07-17)

**Local-first: one owner per event; global is derived.**

```mermaid
flowchart LR
    subgraph repoA [Repo A — .timescope/ committed]
        Aconf[config.json<br/>repo id + Client/Project binding<br/>+ pinned task-types by ID]
        Alog[logs.jsonl<br/>OWNED events]
    end
    subgraph repoB [Repo B — .timescope/ committed]
        Bconf[config.json]
        Blog[logs.jsonl<br/>OWNED events]
    end
    subgraph global [Global store — globalStorage/]
        Reg[registry.json<br/>clients, projects, task_types, repos<br/>all with stable IDs]
        Idx[index.jsonl<br/>DERIVED, rebuildable, disposable]
        Scr[scratch.jsonl<br/>OWNED: non-workspace sessions only]
    end
    Alog -- append-replicate, dedup by ID --> Idx
    Blog -- append-replicate --> Idx
    Aconf -- self-register / cache --> Reg
    Bconf -- self-register --> Reg
    Reg -- Rebuild walks repos --> Idx
```

- **Repo is the authority about itself**: `config.json` (committed) carries repo id, Client/Project binding, and pinned task-types — clone on a new machine → self-registers with the right binding.
- **Registry** lists all entities by stable ID (names display-only → renames are registry-only edits; history follows automatically) plus known repos with paths for efficient rebuild.
- **`index.jsonl` is a cache**: one-directional append replication from repo logs, dedup by ID, fully regenerable via a `Rebuild Global Index` command. Corruption there is never fatal.
- **Timers are independent per window**: each window tracks its own workspace's session; parallel windows on different repos never conflict. Same-workspace double-open = detect + warn.
- **Strict binding**: sessions in a repo always belong to its bound Client/Project; the start picker only asks for task-type. Off-project work → scratch session.
- **Init flow (fixes #2)**: first Start in an unbound folder → "Track time here?" → pick/create Client → Project → write `config.json` + register. No `.timescope` before opt-in.
- **Task-types: one global vocabulary + per-repo pinning.** Schema leaves room for later per-project scoping (`owner_project` filter) with no event rewrites.

## Scope of this issue

1. Registry (`registry.json`, repos only — entity lists arrive with #15), repo `config.json` (binding placeholder), scratch log.
2. Repo-owned appends + one-directional replication into `index.jsonl` (dedup by ID).
3. `TimeScope: Rebuild Global Index` command.
4. Init prompt + opt-in `.timescope` creation.
5. Workspace-scoped runtime state (per-window timers).
6. Legacy: existing global `logs.jsonl` events become global-owned legacy/scratch; job-string→entity mapping deferred to #15.

## Future (not in scope, don't block)

- **Cross-repo quick entry**: from repo A, log a session into repo B — registry knows B's path → append directly, or park in scratch with a `target_repo` field when B is unavailable. Keep schemas compatible.
- Dashboard **Settings panel** (repos view/edit, pins, rebuild button) lands with #6.

## Sequencing

Depends on #42 (newline-safe appends, atomic rewrites, sanitizer). Verified/regression-tested by #47. Build order for the milestone: #42 → this → #47 → #15 → #43 → #6 → #3 → #2.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by Heliman84**
Re-architects storage to **local-first**: one owner per event, with a derived, rebuildable global index. Closes the root cause behind the focus milestone (multi-instance corruption, speculative folders, source confusion). Ships as **v0.6.0**.

Closes #48
Closes #2

## What the user sees
- **Opt-in per folder** — no `.timescope/` is created until you say "Track here" at first Start; an existing `.timescope/` is treated as opted-in. "Never for this folder" is remembered.
- **Open an existing/cloned repo → your jobs are there** (US-06): a repo caches its jobs in `.timescope/config.json` (auto-upgrading an older log-only repo); the Start picker unions them with the global list, so jobs show even on a machine with an empty global.
- **One global dashboard** across every repo + off-project work, reading the derived index.
- **Parallel windows** on different repos track independently without corrupting each other.
- New commands: **Rebuild Global Index**, **Show Storage Status**.

## Model
```mermaid
flowchart LR
    subgraph repo["Repo .timescope/ (committed)"]
        RL["logs.jsonl - OWNED"]
        RC["config.json - repo id + job cache"]
    end
    subgraph global["Global storage (per machine)"]
        SC["scratch.jsonl - OWNED off-project"]
        IX["index.jsonl - DERIVED, rebuildable"]
        RG["registry.json - repos + declines"]
    end
    RL -->|"append + replicate"| IX
    SC -->|replicate| IX
    RG -.->|"rebuild walks repos"| IX
    IX -->|read| DASH["Dashboard"]
```

## Technical
- **One owner per event.** `appendEvent` writes exactly one owned log (workspace when opted-in, else scratch) and replicates into `index.jsonl`. The old global/workspace dual-write is gone.
- **Dashboard reads the derived index**; edits resolve the target in the owned collection and rewrite the owning log, then the index is rebuilt.
- **One-shot migration** of a legacy global `logs.jsonl` into scratch (`.migrated.bak` first, deduped, legacy removed).
- **Registry** caches known repos + per-folder declines (opt-out moved off VS Code state).
- Recovery/session reads dedup by event id across stores (fixes a phantom-open then double-stop bug found in F5).
- New modules: `global_index.ts`, `migration.ts`, `repo_jobs.ts`. Full pure-Node test coverage; `processes.md` storage diagram + `record_format_spec.md` updated.

## Scope boundaries
- Client/Project/Task-type entity model + the job-management/Settings-tab UI go to #15 / #6 (this ships the flat interim). Robust multi-writer concurrency goes to #47.

Dev-log: [docs/dev-log/issue-48-local-first-storage.md](docs/dev-log/issue-48-local-first-storage.md) · Arc: [docs/arc-log/arc-local-first-storage.md](docs/arc-log/arc-local-first-storage.md)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: scripts/seed_legacy_fixture.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// Generate test-workspace-legacy: a v0.2.0-style repo — .timescope/logs.jsonl with
+// jobs but NO config.json, and an EMPTY global store (machine has never seen it).
+// Opening it should auto-upgrade: config.json created + jobs cached (US-06).
+const fs = require("fs"), path = require("path");
+const ROOT = path.resolve("r:/vscode_customizations/timescope");
+const C = path.join(ROOT, "out", "core");
+const { Job } = require(path.join(C, "job"));
+const { Event } = require(path.join(C, "event"));
```
[the remark this answers is not in the document]
Comment (Heliman84) [reply scripts/seed_legacy_fixture.js:8]: Good catch — real portability bug. Fixed in ac7db4b: the repo root is now resolved via `path.resolve(__dirname, "..")` (matching seed_multi_fixture.js), so it runs from anywhere / in CI.

File: src/core/local_opt_in.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+export function enable_local_logging(
+    paths: TimeScopePaths,
+    ws_root: string,
+    ws_name: string,
+    registry_repo: RegistryRepository,
+    now: number
+): string {
+    const ws = workspace_timescope_paths(ws_root);
+    fs.mkdirSync(ws.dir, { recursive: true });
+    const repo_id = ensure_repo_config(ws.config_path);
```
[the remark this answers is not in the document]
Comment (Heliman84) [reply src/core/local_opt_in.ts:80]: Agreed — this is the US-12 edge case. Fixed in ac7db4b: `enable_local_logging` now detects a stray non-directory `.timescope` and throws a clear, actionable error before mkdir; added a test.

File: src/test/test_paths.ts [hunk 3]
```diff
@@ -70,9 +72,53 @@ export function run_resolve_storage_dir_fallback_tests(): void {
 // Exported runner
 // ═══════════════════════════════════════════════════════════════════════════
 
+/**
+ * Tests workspace_timescope_paths:
+ * - Target: workspace_timescope_paths in src/core/paths.ts
+ * - What: derives the `.timescope` dir + jobs/log/config paths under a repo root.
+ * - Why: the opt-in flow (#48) needs these candidate paths without touching disk.
+ */
```
[the remark this answers is not in the document]
Comment (Heliman84) [reply src/test/test_paths.ts:80]: Fixed in ac7db4b — comment target corrected to `src/core/workspace_paths.ts`.

File: docs/record_format_spec.md [hunk 4]
```diff
@@ -1,6 +1,82 @@
...
+A repo's authority about itself, committed with the repository. In 48a it carries only a stable
+repo id (Client/Project binding + pinned task-types arrive with #15):
+
+```json
+{
+  "repo_id": "a1b2c3d4e5f6",
+  "format_version": 1
+}
+```
+
```
[the remark this answers is not in the document]
Comment (Heliman84) [reply docs/record_format_spec.md]: Fixed in ac7db4b — the example now shows `format_version: 2` with the US-06 `jobs` cache, and documents the field.
